RUFAS/biophysical/manure/storage/composting.py

```diff
@@ -0,0 +1,553 @@
+        This function applies the nitrogen losses to the received manure in place.
+
+        Parameters
+        ----------
+        storage_nitrous_oxide_N : float
+            The nitrogen loss through nitrous oxide emissions of the current day, kg.
+        storage_ammonia_N : float
+            The nitrogen loss through ammonia emissions of the current day, kg.
+        storage_N_loss_from_leaching : float
+            The nitrogen loss through leaching of the current day, kg.
+
+        Raises
+        ------
+        ValueError
+            If the total nitrogen losses are greater than the total received manure nitrogen.
+        """
+        received_manure_nitrogen_after_losses = (
+            self._manure_to_process.nitrogen
+            - storage_nitrous_oxide_N
+            - storage_ammonia_N
+            - storage_N_loss_from_leaching
+        )
+        if received_manure_nitrogen_after_losses < 0:
+            self._om.add_error(
+                "Nitrogen loss application error",
+                "Cannot have total nitrogen losses greater than total received manure nitrogen.",
+                info_map={"class": self.__class__.__name__, "function": self._apply_nitrogen_losses.__name__},
+            )
+            raise ValueError(
+                "Nitrogen loss application error: cannot have total nitrogen losses greater than "
+                "total received manure nitrogen."
+            )
+
+        self._manure_to_process.nitrogen = received_manure_nitrogen_after_losses
+
+    @staticmethod
+    def _calculate_composting_ammonia_emissions(
+        composting_type: CompostingType, received_manure_nitrogen: float
+    ) -> float:
+        """
+        This function calculates the total nitrogen loss to ammonia emission of the current day.
+
+        Parameters
+        ----------
+        composting_type : CompostingType
+            The type of composting being used.
+        received_manure_nitrogen : float
+            The nitrogen content of the received manure, kg.
+
+        Returns
+        -------
+        float
+            The total nitrogen loss to ammonia emission of the current day, kg.
+        """
+        return FRACTION_NITROGEN_LOST_TO_AMMONIA_EMISSION[composting_type] * received_manure_nitrogen
+
+    @staticmethod
+    def _calculate_nitrogen_loss_to_leaching(leaching_fraction: float, received_manure_nitrogen: float) -> float:
+        """
+        This function calculates the amount of nitrogen leached out of the manure-bedding
+        pile of the current day.
+
+        Parameters
+        ----------
+        leaching_fraction : float
+            The fraction of nitrogen lost to leaching, unitless.
+        received_manure_nitrogen : float
+            The nitrogen content of the received manure, kg.
+
+        Returns
+        -------
+        float
+            The total nitrogen loss to leaching of the current day, kg.
+        """
+
+        return leaching_fraction * received_manure_nitrogen
```

Comment:
```suggestion
        return FRACTION_NITROGEN_LOST_TO_LEACHING[composting_type] * received_manure_nitrogen
```
Is this needed here? Either functionality-wise or just for consistency with the `def _calculate_composting_ammonia_emissions` syntax

Reply:
Yea I think we can do that here.

Reply:
The only one we can't match in this format is the `storage_nitrous_oxide_n` calculation because that uses the parent Storage `_calculate_nitrogen_loss_to_leaching()` which uses `n2o fraction` (and we wouldn't want to send the composting type to the parent function anyhow).

Reply:
That makes sense and is fine with me!